### PR TITLE
Start of the cleaning of the training scipts + pretrained model

### PR DESCRIPTION
**Pull request by arthurbr11**
Hello!

## Pull Request overview
* Training scripts integration + pretrained models in docs 

## Details
Notably:
* All the scripts for training examples and associate Readme

Still in construction

- Arthur BRESNU

cc @tomaarsen

### CODE REVIEW

File: examples/sparse_encoder/training/ms_marco/README.md [hunk 1]
```diff
@@ -1,2 +1,58 @@
...
+There are **pre-trained models** available, which you can directly use without the need of training your own models. For more information, see: [Pretrained Models](../../../../docs/sparse_encoder/pretrained_models.md).
+
+
+This page describes two strategies to **train a Splade models** on the MS MARCO dataset:
+
+## SparseMultipleNegativesRankingLoss
+**Training code: [train_splade_msmarco_mrl.py](train_splade_msmarco_mrl.py)**
+
+```{eval-rst}
+When we use :class:`~sentence_transformers.sparse_encoder.losses.SparseMultipleNegativesRankingLoss`, we provide triplets: ``(query, positive_passage, negative_passage)`` where ``positive_passage`` is the relevant passage to the query and ``negative_passage`` is a non-relevant passage to the query. We compute the embeddings for all queries, positive passages, and negative passages in the corpus and then optimize the following objective: The ``(query, positive_passage)` pair must be close in the vector space, while ``(query, negative_passage)`` should be distant in vector space.
```
Comment (tomaarsen) [inline examples/sparse_encoder/training/ms_marco/README.md:17]:
```suggestion
When we use :class:`~sentence_transformers.sparse_encoder.losses.SparseMultipleNegativesRankingLoss`, we provide triplets: ``(query, positive_passage, negative_passage)`` where ``positive_passage`` is the relevant passage to the query and ``negative_passage`` is a non-relevant passage to the query. We compute the embeddings for all queries, positive passages, and negative passages in the corpus and then optimize the following objective: The ``(query, positive_passage)`` pair must be close in the vector space, while ``(query, negative_passage)`` should be distant in vector space.
```

File: examples/sparse_encoder/training/ms_marco/README.md [hunk 2]
```diff
@@ -1,2 +1,58 @@
...
+[MS MARCO Passage Ranking](https://github.com/microsoft/MSMARCO-Passage-Ranking) is a large dataset to train models for information retrieval. It consists of about 500k real search queries from Bing search engine with the relevant text passage that answers the query.
+
+This page shows how to **train** Sparse Encoder models, more precisely a Splade model, on this dataset so that it can be used for searching text passages given queries (key words, phrases or questions).
+
+If you are interested in how to use these models, see [Application - Retrieve & Re-Rank](../../applications/retrieve_rerank/README.md).
+
+There are **pre-trained models** available, which you can directly use without the need of training your own models. For more information, see: [Pretrained Models](../../../../docs/sparse_encoder/pretrained_models.md).
+
+
+This page describes two strategies to **train a Splade models** on the MS MARCO dataset:
```
[the remark this answers is not in the document]
Comment (arthurbr11) [reply examples/sparse_encoder/training/ms_marco/README.md:11]: Yes indeed i'm changing it now, I questioned myself to leave it or not as it was in the ST example and ping to my distillation example that use MSMARCO but i feel it's gonna feel duplicate. So for this page i will only keep one.

**Comment by tomaarsen** [comment]
Btw, I trained the MS MARCO one as well: https://huggingface.co/tomaarsen/splade-distilbert-base-uncased-msmarco-mnrl

File: examples/sparse_encoder/training/distillation/README.md [hunk 3]
```diff
@@ -1,2 +1,22 @@
...
+
+Knowledge distillation allows us to compress knowledge from larger, more computationally expensive models (teacher models) into smaller, more efficient sparse models (student models). This approach can leverage bigger model results, including non-sparse models like Cross-Encoders and dense bi-encoders, to compress the knowledge into our small sparse model while maintaining much of the original performance.
+
+## MarginMSE
+**Training code: [train_splade_msmarco_margin.py](train_splade_msmarco_margin.py)**
+
+```{eval-rst}
+:class:`~sentence_transformers.sparse_encoder.losses.SparseMarginMSELoss` is based on the paper of `Hofstätter et al <https://arxiv.org/abs/2010.02666>`_. Like when training with :class:`~sentence_transformers.sparse_encoder.losses.SparseMultipleNegativesRankingLoss`, we can use triplets: ``(query, passage1, passage2)``. However, in contrast to :class:`~sentence_transformers.losses.MultipleNegativesRankingLoss`, `passage1` and `passage2` do not have to be strictly positive/negative, both can be relevant or not relevant for a given query.
```
Comment (tomaarsen) [inline examples/sparse_encoder/training/distillation/README.md:11]: Another difference is the label, should we mention that here already?

File: examples/sparse_encoder/training/distillation/README.md [hunk 4]
```diff
@@ -1,2 +1,22 @@
...
+
+The distillation process works by transferring knowledge from a powerful teacher model (like a Cross-Encoder ensemble) to our efficient sparse encoder student model. We compute the `Cross-Encoder <../../../cross_encoder/applications/README.html>`_ score for ``(query, passage1)`` and ``(query, passage2)`` using the teacher model. We provide scores for 160 million such pairs in our `msmarco-hard-negatives dataset <https://huggingface.co/datasets/sentence-transformers/msmarco-hard-negatives>`_, which contains pre-computed scores from a BERT ensemble Cross-Encoder. We then compute the distance: ``CE_distance = CEScore(query, passage1) - CEScore(query, passage2)``.
```
Comment (tomaarsen) [inline examples/sparse_encoder/training/distillation/README.md:13]: Let's link to this one, it's simpler to use I think: https://huggingface.co/datasets/sentence-transformers/msmarco-scores-ms-marco-MiniLM-L6-v2
(it's the exact same data, just uploaded to a new dataset instead of being hidden as a pickle file)

File: examples/sparse_encoder/training/distillation/README.md (hunk 3 above)
Comment (arthurbr11) [reply examples/sparse_encoder/training/distillation/README.md:11]: I feel like as we are mentioning the score after it make sense to not mention it here. But i'm not against it. What do you think ?

Comment (arthurbr11) [reply examples/sparse_encoder/training/distillation/README.md:11]: That sounds great for me. I get more now what changes you wanted, thanks !

**Comment by arthurbr11** [comment]
Look nice to me as it is, let me know if it's the same for you and i will merge it !

**Review by tomaarsen — APPROVED** [review]
Looks good. We can merge it I think
